src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -469,7 +469,8 @@ public virtual SourceTextContainer Container
 
         internal void CheckSubSpan(TextSpan span)
         {
-            if (span.Start < 0 || span.Start > this.Length || span.End > this.Length)
+            // TextSpan guarantees that Start >= 0 and that Start <= End.
```

Comment:
>// TextSpan guarantees that Start >= 0 and that Start <= End. [](start = 12, length = 61)

We usually use asserts to document and enforce assumptions like this.

Reply:
Thanks, done!